hypothesis-python/src/hypothesis/searchstrategy/types.py

```diff
@@ -177,6 +177,25 @@ def from_typing_type(thing):
         # such as Container
         for t in (typing.KeysView, typing.ValuesView, typing.ItemsView):
             mapping.pop(t, None)
+    if len(mapping) > 1:
+        # issubclass treats bytestring as a kind of sequence, which it is,
+        # but treating it as such breaks everything else when it is presumed
+        # to be a generic sequence or container that could hold any item.
+        # Except for sequences of integers, or unions which include integer!
```

Comment:
It's still not clear to me why unions are the only thing we need to special case here. Why is a request for a `Sequence[object]` not adequately fulfilled by `ByteString`?

Reply:
It would be... so instead of matching the exact type, we check if int is a subtype of the element type (or any part of the union). Makes sense.